perun-wui-registrar/src/main/java/cz/metacentrum/perun/wui/registrar/pages/steps/SummaryStep.java

```diff
@@ -867,6 +867,7 @@ public void onClick(ClickEvent event) {
 				button.addClickHandler(new ClickHandler() {
 					@Override
 					public void onClick(ClickEvent event) {
+						continueButton.addClickHandler(clickHandler);
```

Comment:
I'm not sure what this is supposed to do. Do you wish to perform redirect once you close the modal window? Because I think now you can try it multiple times and it always checks the current state of the submitted application and either shows the modal again or performs the redirect by passing click event to the default handler: `clickHandler.onClick(event);`.
Your code would simply add the handler to the button, hence when clicked again, it would show the modal window and perform the redirect together.

Reply:
What I want to achieve is that even when the application is not processed yet, the user can be redirected to the target location, as the location might not be the service itself, just some "portal" with information. 
Perhaps an alternative and better solution would be to add another button to the modal window like "I understand and want to proceed" next to the "I understand" button.